budcamp/Homework3/Homework3/task2/TestContainer.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Homework3
+{
+    internal class TestContainer
+    {
+        private Dictionary<string, List<Test>> dictOfTests;
+
+        public TestContainer()
+        {
+            dictOfTests = new Dictionary<string, List<Test>>();
+        }
+
+        public void AddTest(Test test)
+        {
+            if (!dictOfTests.ContainsKey(test.Topic))
+            {
+                dictOfTests[test.Topic] = new List<Test>();
+                dictOfTests[test.Topic].Add(test);
+            }
+            else
+            {
+                Console.WriteLine("this test is alredy exists or ID reserved");
```

Comment:
This doesn't make sense, again why you have so much `Console.WriteLine`s in your code.
I think what you wanted to do is the following:
```
if (!dictOfTests.ContainsKey(test.Topic))
{
    dictOfTests[test.Topic] = new List<Test>();
}
dictOfTests[test.Topic].Add(test);
```